ouroboros-consensus/src/ouroboros-consensus/Ouroboros/Consensus/Mempool/Update.hs

```diff
@@ -217,9 +205,25 @@ pureTryAddTx cfg txSize wti tx is
   | otherwise
   = NoSpaceLeft
     where
-      (eVtx, vr) = extendVRNew cfg txSize wti tx $ validationResultFromIS is
+      (eVtx, vr) = extendVRNew cfg wti tx $ validationResultFromIS is
       is'        = internalStateFromVR vr
 
+willItFit ::
+     Measure (TxMeasure blk)
+  => InternalState blk -> TxMeasure blk -> Bool
+willItFit is tx
+  | not (tx Measure.<= cap) = True   -- let it be found invalid
+  | otherwise               =
+    go (isTxs is) (max 0 $ isMultiplicity is - 1)
+  where
+    cap = isCapacity is
+
+    go !txseq = \case
+      0 -> msSize (TxSeq.toMempoolSize txseq) `Measure.plus` tx Measure.<= cap
+      n -> case txseq of
+        TxSeq.Empty -> go txseq 0
```

Comment:
When `txseq` is `TxSeq.Empty`: As we already established `tx Measure.<= cap` after the first guard above, we could also directly return `True` here as
```haskell
  go txseq 0
≡ msSize (TxSeq.toMempoolSize txseq) `Measure.plus` tx Measure.<= cap
≡ Measure.zero `Measure.plus`                       tx Measure.<= cap
≡                                                   tx Measure.<= cap
≡ True
```